### PR TITLE
Add typescript support to Portal component

### PR DESCRIPTION
**Pull request by PlayMa256**
Adds typescript support to `Portal` component. I've tried to follow the props available on the docs and on the `Base` component as well (since ts started to complain about it).

Let me know if any changes are required!

### CODE REVIEW

**Comment by PlayMa256** [comment]
Tests seems to be failing due to a hash that changed on the classname (totally unrelated to this pr).

File: yarn.lock [hunk 1]
```diff
@@ -1896,16 +1811,14 @@ babel-plugin-jest-hoist@^23.2.0:
   resolved "https://registry.yarnpkg.com/babel-plugin-jest-hoist/-/babel-plugin-jest-hoist-23.2.0.tgz#e61fae05a1ca8801aadee57a6d66b8cefaf44167"
 
 babel-plugin-styled-components@^1.2.0:
-  version "1.5.1"
-  resolved "https://registry.yarnpkg.com/babel-plugin-styled-components/-/babel-plugin-styled-components-1.5.1.tgz#31dbeb696d1354d1585e60d66c7905f5e474afcd"
+  version "1.6.3"
```
Comment (diegohaz) [inline yarn.lock:1814]: I guess hashes were changed because of this. That's ok. You can update snapshots (`yarn test -u` I think)

**Comment by PlayMa256** [comment]
Everything seems good now

**Comment by diegohaz** [comment]
Thanks a lot for contributing!

Based on our [community guidelines](https://github.com/reakit/reakit/blob/master/CONTRIBUTING.md), every person who has a PR of any kind merged is offered **an invitation to the ReaKit organization**.

Should you accept, you'll get write access to the main repository and a fancy [ReaKit logo](https://github.com/reakit/reakit/tree/master/logo) on your GitHub profile. You'll be able to label and close issues, create new branches etc. Make sure to read our [contribution and community guidelines](https://github.com/reakit/reakit/blob/master/CONTRIBUTING.md), which explains all of this in more detail.

If you have any questions, let me know!

**Pull request merged**